src/components/TestFileView/index.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { ReactNode, useState } from 'react';
+import styled from 'styled-components';
+import * as Accordion from '@radix-ui/react-accordion';
+import { AccordionIcon } from '../../icons';
+import { FailedIcon } from '../../icons';
+
+interface Props {
+  children: ReactNode;
+  'data-state'?: string;
+  failed?: string[];
+  value?: string;
+}
+
+const TestFileView: React.FC = () => {
+  const [failed, setFailed] = useState(['item-1', 'item-2']);
```

Comment:
From the design, I understood that it has a lot of states to handle.
We can use redux for state management. Will be helpful for scalability.

Reply:
As discussed on the call, we will be using [React Context API ](https://beta.reactjs.org/blog/2018/03/29/react-v-16-3#official-context-api)